imap_processing/mag/l0/decom_mag.py

```diff
@@ -39,23 +42,95 @@ def decom_packets(packet_file_path: str) -> list[MagL0]:
     data_list = []
 
     with open(packet_file_path, "rb") as binary_data:
```

Comment:
Could you use the decom.decom_packets function here?

Reply:
I'd have to loop through twice. Once to get the list of decommed packets from decom_packets, then again to go through that list and apply the MagL0 code to it. One alternative might be to write an optional parameter in decom_packets that includes a dataclass to apply ... Since Python's typing system is so loose, you could then return either a list of packets or a list of, for example, MagL0 objects. Might be worth doing now that more people are using the dataclasses...?

Reply:
You wouldn't know what APIDs are in the file to pass in a data class though, so you'd probably still have to loop through twice. For example, if the file contains science and housekeeping packets, you would need to pass in multiple data classes and have a way to match up the APIDs. I might be missing something here though. But I do see what you're saying about the way you're doing it now avoids the need to loop through the packets twice.

Reply:
Ah yeah... that's a good point. You could probably do it with some nasty dictionary shenanigans... but I hate nasty dictionary shenanigans! Personally, I think it's fine to just have every instrument do the opening and reading themselves, since there are special cases (dataclasses) and weird scenarios (what you said)